flowmachine/flowmachine/core/cache.py

```diff
@@ -801,7 +804,9 @@ def cache_table_exists(connection: "Connection", query_id: str) -> bool:
 
         try:
             await asyncio.wait_for(
-                asyncio.get_running_loop().run_in_executor(pool, shrink_func),
+                asyncio.get_running_loop().run_in_executor(
+                    pool, copy_context().run, shrink_func
```

Comment:
Might be worth adding a comment here to explain what the `copy_context` incantation is about?